### PR TITLE
API: New entity "status"

### PR DESCRIPTION
**Pull request by annando**
This adds support for the endpoint `/api/v1/timelines/public`. The entity is not really complete, but it is a start.

### CODE REVIEW

File: src/Module/Api/Mastodon/Timelines/PublicTimeline.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model\Item;
+use Friendica\Module\BaseApi;
+use Friendica\Network\HTTPException;
+
+/**
+ * @see https://docs.joinmastodon.org/methods/timelines/
+ */
+class PublicTimeline extends BaseApi
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Timelines/PublicTimeline.php:34]: Please rename to just `Public`. Don't forget to rename the file as well.
```suggestion
class Public extends BaseApi
```

File: src/Object/Api/Mastodon/Status.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ * Class Status
+ *
+ * @see https://docs.joinmastodon.org/entities/status
+ */
+class Status extends BaseEntity
+{
+	/** @var string */
+	protected $id;
+	/** @var string (Datetime) */
+    protected $created_at;
```
Comment (MrPetovan) [inline src/Object/Api/Mastodon/Status.php:37]: Code standards: Please use tabs for indentation.

File: src/Object/Api/Mastodon/Status.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		$this->favourites_count = 0;
+		$this->favourited = false;
+		$this->reblogged = false;
+		$this->muted = false;
+		$this->bookmarked = false;
+		$this->pinned = false;
+		$this->content;
+		$this->reblog = null;
+		$this->application = null;
+		$this->account         = $account->toArray();
```
Comment (MrPetovan) [inline src/Object/Api/Mastodon/Status.php:127]: Code standards: Please decide whether you want to align equal signs or not.

File: static/routes.config.php [hunk 4]
```diff
@@ -48,12 +48,13 @@
...
-			'/follow_requests/{id:\d+}/{action}' => [Module\Api\Mastodon\FollowRequests::class, [        R::POST]],
-			'/instance'                          => [Module\Api\Mastodon\Instance::class,       [R::GET         ]],
-			'/instance/peers'                    => [Module\Api\Mastodon\Instance\Peers::class, [R::GET         ]],
+			'/custom_emojis'                     => [Module\Api\Mastodon\CustomEmojis::class,             [R::GET         ]],
+			'/directory'                         => [Module\Api\Mastodon\Directory::class,                [R::GET         ]],
+			'/follow_requests'                   => [Module\Api\Mastodon\FollowRequests::class,           [R::GET         ]],
+			'/follow_requests/{id:\d+}/{action}' => [Module\Api\Mastodon\FollowRequests::class,           [        R::POST]],
+			'/instance'                          => [Module\Api\Mastodon\Instance::class,                 [R::GET         ]],
+			'/instance/peers'                    => [Module\Api\Mastodon\Instance\Peers::class,           [R::GET         ]],
+			'/timelines/public'                  => [Module\Api\Mastodon\Timelines\PublicTimeline::class, [R::GET         ]],
```
Comment (MrPetovan) [inline static/routes.config.php:57]:
```suggestion
			'/custom_emojis'                     => [Module\Api\Mastodon\CustomEmojis::class,     [R::GET         ]],
			'/directory'                         => [Module\Api\Mastodon\Directory::class,        [R::GET         ]],
			'/follow_requests'                   => [Module\Api\Mastodon\FollowRequests::class,   [R::GET         ]],
			'/follow_requests/{id:\d+}/{action}' => [Module\Api\Mastodon\FollowRequests::class,   [        R::POST]],
			'/instance'                          => [Module\Api\Mastodon\Instance::class,         [R::GET         ]],
			'/instance/peers'                    => [Module\Api\Mastodon\Instance\Peers::class,   [R::GET         ]],
			'/timelines/public'                  => [Module\Api\Mastodon\Timelines\Public::class, [R::GET         ]],
```

File: src/Module/Api/Mastodon/Timelines/PublicTimeline.php (hunk 1 above)
Comment (annando) [reply src/Module/Api/Mastodon/Timelines/PublicTimeline.php:34]: `Public` is a reserved word.

File: src/Object/Api/Mastodon/Status.php (hunk 2 above)
Comment (annando) [reply src/Object/Api/Mastodon/Status.php]: Done

File: src/Object/Api/Mastodon/Status.php (hunk 3 above)
Comment (annando) [reply src/Object/Api/Mastodon/Status.php]: Done

File: src/Object/Api/Mastodon/Status.php (hunk 2 above)
Comment (MrPetovan) [reply src/Object/Api/Mastodon/Status.php]: Please check the whole file, I didn't add a comment everywhere spaces were used.

File: src/Module/Api/Mastodon/Timelines/PublicTimeline.php (hunk 1 above)
Comment (MrPetovan) [reply src/Module/Api/Mastodon/Timelines/PublicTimeline.php:34]: Ah yes, too bad.

File: src/Object/Api/Mastodon/Status.php (hunk 2 above)
Comment (annando) [reply src/Object/Api/Mastodon/Status.php]: Done

**Pull request merged**